wp/lib/bap_wp/src/precondition.ml

```diff
@@ -398,12 +399,21 @@ let subst_fun_outputs ?tid_name:(tid_name = "") ~inputs:(inputs : Var.t list)
       Env.add_call_pred env sub_to) in
   Constr.substitute post subs_from subs_to, env
 
+(* Creates fresh names for [vars] found in the constraint. Optionally takes in
+   a function that maps the original variable name to the fresh name. Defaults
+   to [fresh_var]. *)
+let freshen ?(name = Format.sprintf "fresh_%s") (constr : Constr.t)
```

Comment:
I wonder if this should be in `environment.ml`; it seems useful.

Reply:
Can move it there!